Sources/_SwiftSyntaxTestSupport/SourceEditsTestUtils.swift

```diff
@@ -0,0 +1,76 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import XCTest
+
+/// Returns the `ConcurrentEdits`s to transition from `base` to `new`.
+public func getConcurrentEdits(from base: String, to new: String) -> ConcurrentEdits {
+  let diffCollection = new.difference(from: base)
+
+  let diffArray: [(offset: Int, isInsert: Bool)] =
+    diffCollection
+    .map { diff in
+      switch diff {
+      case .remove(offset: let offset, element: _, associatedWith: _):
+        return (offset: offset, isInsert: false)
+      case .insert(offset: let offset, element: _, associatedWith: _):
+        return (offset: offset, isInsert: true)
+      }
+    }
+    .sorted {
+      // Change.remove should prior to Change.insert
+      if $0.offset < $1.offset {
+        return true
+      } else if $0.offset == $1.offset {
+        return !$0.isInsert
+      } else {
+        return false
+      }
+    }
+
+  let sourceEdits = diffArray.map({
+    return $0.isInsert ? SourceEdit(offset: $0.offset, length: 0, replacementLength: 1) : SourceEdit(offset: $0.offset, length: 1, replacementLength: 0)
```

Comment:
I would make this an `if` statement. It’s just easier to read:

```suggestion
    if $0.isInsert {
      return SourceEdit(offset: $0.offset, length: 0, replacementLength: 1)
    } else {
      SourceEdit(offset: $0.offset, length: 1, replacementLength: 0)
    }
```

---

Also, I think for the future we want this to be able to merge `SourceEdit`s to create `SourceEdit`s that either have `length` or `replacementLength` that’s created.

I.e. `base = "0123"` and `new = "abc0123"` should create a single `SourceEdit(offset: 0, length: 0, replacementLength: 3)`.

Reply:
I think for the incremental parsing test, we could just merge them into a `ConcurrentEdits` to help us test. Do you have some use cases about merging `SourceEdit`s into a big one?

Reply:
Oh, I didn’t realize that the conversion into `ConcurrentEdits` merges them. In that case, it’s good, I think.